feat: Create UI for import to cart page

Parthlakhani opened the pull request:
closes #11525

Parthlakhani committed:
feat: Create import entries and import entries link component

Pucek9 commented on feature-libs/cart/import-export/components/import-to-cart/import-entries-link/import-entries-link-component.html:
```diff
@@ -0,0 +1,3 @@
+<button class="cx-btn btn btn-block btn-primary col-lg-6">
+  Import Products
```
translation :)

Pucek9 commented on feature-libs/cart/import-export/components/import-to-cart/import-entries-link/import-entries-link-component.html:
```diff
@@ -0,0 +1,3 @@
+<button class="cx-btn btn btn-block btn-primary col-lg-6">
```
are all these bootstrap classes necessary? btw it should be link instead of button

Pucek9 commented on feature-libs/cart/import-export/styles/_import-entries.scss:
```diff
@@ -0,0 +1,9 @@
+cx-import-entries-link cx-import-entries {
```
I don't think so we should group components for adding styles to separated elemetns

Parthlakhani committed:
feat: Create import to cart UI (GH-11525)

Created model for import to cart UI.
Added launchDialog service for using dialog.
Replaced import to cart link with button.

Pucek9 committed:
uncle Michał's Improvements

Pucek9 committed:
revert property name, use import service directly

Pucek9 committed:
no longer necessary change

Pucek9 committed:
working on file instead of file list

Parthlakhani committed:
refactor: Change translation

Parthlakhani committed:
add support for filesize from default config

Pucek9 commented on feature-libs/cart/import-export/assets/translations/en/import-export.i18n.ts:
```diff
@@ -2,6 +2,28 @@ export const exportEntries = {
+  importProductsSubtitle:
+    'Add products by importing a .CSV file and creating a new saved cart.',
+  importProductFileText:
+    'The text file should list the product SKUs and quantities, separated by a comma. File size should not exceed ',
```
```suggestion
    'The text file should list the product SKUs and quantities, separated by a comma. File size should not exceed {{ size }} kb',
```

Pucek9 commented on feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html:
```diff
@@ -0,0 +1,126 @@
+          {{ 'importEntriesDialog.importProductsSubtitle' | cxTranslate }}
+        </p>
+        <p>
+          {{ 'importEntriesDialog.importProductFileText' | cxTranslate }}
+          {{ fileSize }}{{ 'importEntriesDialog.kb' | cxTranslate }}
```
```suggestion
          {{ 'importEntriesDialog.importProductFileText' | cxTranslate: { size: fileSize } }}
```

Pucek9 commented on feature-libs/cart/import-export/assets/translations/en/import-export.i18n.ts:
```diff
@@ -2,6 +2,28 @@ export const exportEntries = {
+  charactersLeft: 'characters left: {{count}}',
+  cancel: 'Cancel',
+  upload: 'Upload',
+  kb: 'KB.',
```
```suggestion
```

Parthlakhani committed:
Update feature-libs/cart/import-export/components/import-to-cart/import-entries-dialog/import-entries-dialog.component.html

Co-authored-by: Michał Gruca <pucek9@gmail.com>

Parthlakhani committed:
Update feature-libs/cart/import-export/assets/translations/en/import-export.i18n.ts

Co-authored-by: Michał Gruca <pucek9@gmail.com>

Parthlakhani committed:
Update feature-libs/cart/import-export/assets/translations/en/import-export.i18n.ts

Co-authored-by: Michał Gruca <pucek9@gmail.com>

Parthlakhani committed:
Merge branch 'develop' of https://github.com/Parthlakhani/spartacus into feature/GH-11525

Parthlakhani committed:
Merge branch 'epic/cart-import-export' into feature/GH-11525

Parthlakhani committed:
update UI and fix importExportConfig

Pucek9 committed:
resolve conflict

Pucek9 committed:
up

Pucek9 committed:
remove import entries launch dialog service

Pucek9 committed:
prettier:fix

Pucek9 committed:
avoid build errors

Pucek9 committed:
add optional sign

Pucek9 committed:
translatable aria labels

Pucek9 committed:
up

Pucek9 committed:
up

Pucek9 committed:
up

Parthlakhani committed:
remove unnecessary files

Pucek9 committed:
Merge branch 'epic/cart-import-export' of https://github.com/SAP/spartacus into feature/GH-11525

Matejk00 commented on projects/storefrontapp/src/app/spartacus/spartacus-features.module.ts:
```diff
@@ -192,7 +191,6 @@ if (environment.cpq) {
 
     /************************* External features *************************/
     UserFeatureModule,
-    CheckoutFeatureModule,
```
Should we delete this one?
